cocos/network/WebSocket.h

```diff
@@ -238,7 +239,7 @@ class CC_DLL WebSocket
 
     struct lws*         _wsInstance;
     struct lws_context* _wsContext;
-    std::shared_ptr<bool> _isDestroyed;
+    std::shared_ptr<std::atomic<bool>> _isDestroyed;
```

Comment:
> You could refer to WebSocket::onConnectionError method.
If it's a raw type like bool or std::atomic<bool>, lambda will just allocate a new variable to from _isDestroyed, the lifecycle of the new captured variable will not be the same as WebSocket::_isDestroyed. In this situation, although the _isDestroyed is set to true, the lambda will still catch a false value.

I think it is because you pass value not reference to lambda.